src/main/scala/esmeta/analyzer/TypeAnalyzer.scala

```diff
@@ -101,6 +102,48 @@ class TypeAnalyzer(
         case _ =>
           super.doCall(callerNp, callerSt, calleeFunc, args, captured, method)
 
+    /** call transition */
+    override def getCalleeEntries(
+      callerNp: NodePoint[Call],
+      callerSt: AbsState,
+      calleeFunc: Func,
+      args: List[AbsValue],
+      captured: Map[Name, AbsValue],
+      method: Boolean,
+    ): List[(NodePoint[_], AbsState)] =
+      for {
+        baseView <- if (TY_SENS) getViewFromArgs(args) else List(View())
+      } yield {
+        // handle ir callsite sensitivity
+        val NodePoint(callerFunc, callSite, callerView) = callerNp
+
+        val calleeNp = NodePoint(calleeFunc, calleeFunc.entry, baseView)
+        val calleeSt = callerSt.copied(locals =
+          getLocals(
+            CallPoint(callerNp, calleeNp),
+            args,
+            cont = false,
+            method,
+          ) ++ captured,
+        )
+        (calleeNp, calleeSt)
+      }
+
+    private def getViewFromArgs(args: List[AbsValue]): List[View] =
+      val types = getTypes(args.map(_.ty))
+      val views = types.map(t => View().copy(tys = t))
+      views
+
+    private def getTypes(args: List[ValueTy]): List[List[ValueTy]] = {
+      args.foldRight(List(List[ValueTy]())) {
```

Comment:
Is there any reason for choosing to use foldRight?

Reply:
No, just copied from JSTAR. Maybe a performance issue?

Reply:
I think foldRight is the only correct option because we iteratively attach arguments on left side of the list